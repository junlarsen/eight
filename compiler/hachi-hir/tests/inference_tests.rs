mod common;

macro_rules! inference_test {
    ($name:ident, $src:expr) => {
        #[test]
        fn $name() {
            use hachi_hir::format::HirModuleFormatter;

            let path = format!("{}/tests/{}", env!("CARGO_MANIFEST_DIR"), $src);
            let src = std::fs::read_to_string(path).unwrap();
            let module = assert_hir_module_infers!(&src);
            let doc = HirModuleFormatter::format_hir_module_to_string(&module);
            insta::assert_snapshot!(doc);
        }
    };
}

inference_test!(
    test_inference_uninitialized_inference,
    "data/inference/uninitialized_inference.test"
);
inference_test!(
    test_inference_local_resolution,
    "data/inference/local_resolution.test"
);
inference_test!(
    test_inference_chaining_infers_unit,
    "data/inference/chaining_infers_unit.test"
);
