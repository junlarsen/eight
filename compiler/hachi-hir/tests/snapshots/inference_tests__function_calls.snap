---
source: compiler/hachi-hir/tests/inference_tests.rs
expression: doc
snapshot_kind: text
---
module {
  // module intrinsic scalar types
  intrinsic_scalar bool;
  intrinsic_scalar i32;
  intrinsic_scalar unit;
  
  // module intrinsic functions
  
  
  // module types
  
  
  // module functions
  fn bar(x: i32) -> i32 {
    return (x as i32);
  }
  fn foo() -> unit {
    let k: i32 = ((bar as fn((i32))->i32)::<>((123 as i32)) as i32);
    let z: i32 = ((generic as fn((i32))->i32)::<>((k as i32)) as i32);
    let z: bool = ((generic as fn((bool))->bool)::<>((true as bool)) as bool);
  }
  fn generic(x: $15) -> $15 {
    return (x as i32);
  }
}
